### ISSUE TITLE
[Arch/Manjaro] Kernel Panic with 6.10.0-3

### ISSUE THREAD

**Issue by croci** [craigcabrey/thinkbook-13x-gen4-enablement#8]
First let me say that your work is amazing! My laptop essentially only works thanks to you and you saved me so much time. Thanks a lot!

I'm using Manjaro and so far all your fixes have carried through seamlessly. However, ever since the kernel was updated to 6.10.0-3 the laptop starts randomly having kernel panics (blinking caps lock and complete freeze). Did you experience this as well by any chance? I guess it is only a matter of waiting for the next stable kernel.

If this is new to you, please delete this issue. Thanks a lot again for all your help!

**Comment by craigcabrey** [comment]
I have not. The -3 on your version implies there were several Manjaro release of that kernel. Have you looked at what patches they are applying? Or perhaps try a vanilla release of 6.10?

**Comment by hxss** [comment]
Hi guys. Same issue on 6.10. I'm still not found source of problem and just test some kernel params

**Comment by craigcabrey** [comment]
It has been completely stable on Fedora, so there either something that Fedora is patching to fix things, or Arch is applying a patch which breaks things. 

It's probably the latter. Please try a vanilla kernel.

**Comment by rcvd** [comment]
I installed the notebook with the 6.10.2-arch1-1.1 kernel on my ArchLinux system using the ALHP Repository with v3 enhanced package builds (https://somegit.dev/ALHP/ALHP.GO). I can confirm that my laptop has been running for over a week without a kernel panic.

**Comment by hxss** [comment]
small updates:
- `6.10.1-arch1-1 with ahci.mobile_lpm_policy=1` panic
- `6.10.3-zen1-1` panic
- `6.10.3-arch1-1` panic

all kernels runs with `i8042.nopnp=1 i915.enable_psr=0 nmi_watchdog=0`. Also tested with enabled nmi_watchdog - still panics.

**Comment by rcvd** [comment]
@hxss: Have you tried the ALHP repository?

**Comment by hxss** [comment]
- `6.10.2-hardened1-2-hardened` panic
- `6.11-mainline-rc2-1` panic

**Comment by hxss** [comment]
@rcvd I enabled ALHP v3 repos and update the system, but pacman cant find and install `core-x86-64-v3/linux core-x86-64-v3/linux-headers`

**Comment by hxss** [comment]
- `core-x86-64-v3/linux 6.10.3.arch1-2.1` panic
- `cachyos-v3/linux-cachyos 6.10.3-3` panic

**Comment by rcvd** [comment]
@hxss: I built my own 6.9.10 kernel with the sound fix from 6.10 after getting kernel panics again with the latest releases - this is running rock-stable. Do you want me to put it somewhere for you to download? You can pin it in /etc/pacman.conf so it won't get updated until you explicitly force it.

**Comment by hxss** [comment]
@rcvd yeah please create instructions in gist. I'll try your method. As I understand you and @craigcabrey works stable on 6.10?

**Comment by hxss** [comment]
- `core-x86-64-v3/linux 6.10.3.arch1-2.1` without `i915.enable_psr=0` panic. But I notice smooth graphic work, so I remove this parameter from all boot points

**Comment by kmasterycsl** [comment]
Here is my experience
- 6.10.3-2 is rock solid stable on Fedora 40
- 6.10 - 6.11 are always panic on Manjaro after few minutes with cap lock blinking

**Comment by rcvd** [comment]
- Fedora is working stable with 6.10 and even 6.11 (Rawhide)
- Arch worked on 6.10 with ALHP repository but stopped with the latest updates (random kernel panics with blinking caps lock), so I added the sound patch from 6.10 to 6.9 and it is stable since then

**Comment by croci** [comment]
I wouldn't know where to start, but is there a way to pin the causes down and/or open a bug report? If it works in Fedora (but not with the mainline Linux kernels, correct?) then the solution is already out there.

**Comment by rcvd** [comment]
6.9.10 PKGBUILD with sound patch for ThinkBook 13x gen4: https://github.com/rcvd/thinkbook-arch

**Comment by craigcabrey** [comment]
You could try installing a Fedora kernel on your Arch/Manjaro install. Definitely more advanced but not impossible. Everything is rock solid on the Fedora side.

You can look through the patches maintained by the Fedora kernel too.

**Comment by craigcabrey** [comment]
This is the patch that is applied to the fedora 40 kernel (6.10.3 as of this writing): https://src.fedoraproject.org/rpms/kernel/blob/f40/f/patch-6.10-redhat.patch

**Comment by rcvd** [comment]
It looks like the latest kernel on Archlinux (Linux version 6.10.4-arch2-1) fixed the kernel panics.

**Comment by hxss** [comment]
in my case 6.10.4-arch2-1 panics too. Also panics:
- linux-zen 6.10.4.zen2-1.1
- linux-cachyos 6.10.5-2 (6.10.4 panics too)
- linux-clear 6.10.5-1

@rcvd Following your instructions, I downgraded to 6.9.10 with sound patch and its first 24h of stable work. Thank you.

**Comment by allflame** [comment]
For me blacklisting ipu6 module `blacklist intel-ipu6` solves the random panic issue (6.10.5-1-default, from Suse Tumbleweed). Camera still doesn't work with `libcamera 0.3.1`, so not much of an issue at this stage

**Comment by TenzinJamyangZHS** [comment]
> For me blacklisting ipu6 module `blacklist intel-ipu6` solves the random panic issue (6.10.5-1-default, from Suse Tumbleweed). Camera still doesn't work with `libcamera 0.3.1`, so not much of an issue at this stage

Hi, what else not work on this laptop with opensuse tumbleweed? And how about battery life? And energy-efficient kernels?

**Comment by hxss** [comment]
I can confirm that without `intel_ipu6` latest `6.11.0-5-cachyos` kernel works stable. Also `6.9.10` that I used before dont have the module.

**Comment by croci** [comment]
I am on Fedora 41 and panics also occur. @craigcabrey do you experience the same issue?

**Comment by croci** [comment]
On Fedora 41 installing the RPM Fusion Intel IPU6 drivers fixes the kernel panics without need for blacklisting the module. Not sure if this is applicable to other distros.

**Comment by craigcabrey** [comment]
Can anyone confirm this is still happening in Arch/Manjaro? Otherwise we can close this out
